lib/segment/src/index/hnsw_index/graph_links/serializer.rs

```diff
@@ -34,15 +39,38 @@ enum Kind {
     Compressed {
         compressed_offsets: Vec<u8>,
         offsets_parameters: bitpacking_ordered::Parameters,
+        quantized_vector_size: Option<usize>,
     },
 }
 
 impl GraphLinksSerializer {
-    pub fn new(
+    pub fn new<'a>(
+        edges: Vec<Vec<Vec<PointOffsetType>>>,
+        format_param: GraphLinksFormatParam<&'a QuantizedVectors>,
+        hnsw_m: HnswM,
+    ) -> OperationResult<Self> {
+        let get_quantized_vector;
+        let format = match format_param {
+            GraphLinksFormatParam::Plain => GraphLinksFormatParam::Plain,
+            GraphLinksFormatParam::Compressed => GraphLinksFormatParam::Compressed,
+            GraphLinksFormatParam::CompressedWithVectors(qv) => {
+                get_quantized_vector = |i| qv.get_quantized_vector(i);
+                GraphLinksFormatParam::CompressedWithVectors(
+                    &get_quantized_vector as &dyn Fn(PointOffsetType) -> &'a [u8],
+                )
+            }
+        };
+
+        Self::new_inner(edges, format, hnsw_m)
+    }
+
+    /// Unwrapped version of [`Self::new`] that accepts `&dyn Fn` rather than
+    /// `&QuantizedVectors`. This unwrapped version is intended for testing.
+    pub(super) fn new_inner<'a>(
         mut edges: Vec<Vec<Vec<PointOffsetType>>>,
-        format: GraphLinksFormat,
+        format_param: GraphLinksFormatParam<&dyn Fn(PointOffsetType) -> &'a [u8]>,
```

Comment:
since we actually want to only have function inside enum and nothing else, let's try to avoid `&dyn Fn` as a generic param